modules/tinymce/src/plugins/accordion/test/ts/browser/AccordionPluginTest.ts

```diff
@@ -121,12 +130,154 @@ describe('browser.tinymce.plugins.accordion.AccordionPluginTest', () => {
+    TinySelections.setCursor(editor, [ 0, 0, 0 ], 0);
+    editor.execCommand('ToggleAccordion', false, false);
+    TinyAssertions.assertContentPresence(editor, { 'details:not([open="open"])': 1 });
+    TinyAssertions.assertCursor(editor, [ 0, 0, 0 ], 0);
+    editor.execCommand('ToggleAccordion', false, true);
+    TinyAssertions.assertContentPresence(editor, { 'details[open="open"]': 1 });
+    TinyAssertions.assertCursor(editor, [ 0, 0, 0 ], 0);
+    editor.execCommand('ToggleAccordion', false, true);
+    TinyAssertions.assertContentPresence(editor, { 'details[open="open"]': 1 });
+    TinyAssertions.assertCursor(editor, [ 0, 0, 0 ], 0);
+  });
+
+  it('TINY-9731: Toggle all accordion elements', () => {
+    const editor = hook.editor();
+    editor.setContent([ createAccordion({ open: true }), createAccordion({ open: true }) ].join(''));
+    TinySelections.setCursor(editor, [ 0, 0, 0 ], 0);
+    editor.execCommand('ToggleAllAccordions');
+    TinyAssertions.assertContentPresence(editor, { 'details:not([open="open"])': 2 });
+    TinyAssertions.assertCursor(editor, [ 0, 0, 0 ], 0);
+    editor.execCommand('ToggleAllAccordions');
+    TinyAssertions.assertContentPresence(editor, { 'details[open="open"]': 2 });
+    TinyAssertions.assertCursor(editor, [ 0, 0, 0 ], 0);
+  });
+
+  it('TINY-9731: Toggle all accordion elements with an argument', () => {
+    const editor = hook.editor();
+    editor.setContent([ createAccordion({ open: true }), createAccordion({ open: true }) ].join(''));
+    TinySelections.setCursor(editor, [ 0, 0, 0 ], 0);
+    editor.execCommand('ToggleAllAccordions', false, false);
+    TinyAssertions.assertContentPresence(editor, { 'details:not([open="open"])': 2 });
+    TinyAssertions.assertCursor(editor, [ 0, 0, 0 ], 0);
+    editor.execCommand('ToggleAllAccordions', true, true);
+    TinyAssertions.assertContentPresence(editor, { 'details[open="open"]': 2 });
+    TinyAssertions.assertCursor(editor, [ 0, 0, 0 ], 0);
+    editor.execCommand('ToggleAllAccordions', true, true);
+    TinyAssertions.assertContentPresence(editor, { 'details[open="open"]': 2 });
+    TinyAssertions.assertCursor(editor, [ 0, 0, 0 ], 0);
+  });
+
+  it('TINY-9731: Emit the "ToggledAccordion" event', () => {
+    const editor = hook.editor();
+    editor.setContent(createAccordion({ summary: 'tiny' }));
+    TinySelections.setCursor(editor, [ 0, 0, 0 ], 'tiny'.length);
+    testEvent(editor, 'ToggledAccordion', 'ToggleAccordion', (event: ToggledAccordionEvent) => {
+      assert.equal(event.element.nodeName, 'DETAILS');
+      assert.isFalse(event.state);
+    });
+    testEvent(editor, 'ToggledAccordion', 'ToggleAccordion', (event: ToggledAccordionEvent) => {
+      assert.equal(event.element.nodeName, 'DETAILS');
+      assert.isTrue(event.state);
+    });
+  });
+
+  it('TINY-9731: Emit the "ToggledAllAccordions" event', () => {
+    const editor = hook.editor();
+    editor.setContent(createAccordion({ summary: 'tiny' }));
```

Comment:
add a case for several accordions to be in the content

Reply:
Updated the test with 2 accordions. Added state checks as well.